### DISCUSSION TITLE
Did TikTok-Api use Selenium to scrape Tiktok's data?

### DISCUSSION THREAD

**Discussion by novemberdude** [davidteather/TikTok-Api#428] [Q&A] [answered]
Hey David, I'm wondering a question. In the module tiktok.py and in the getData(self) method, I saw that you create a Request with an url like `https://m.tiktok.com/api/item_list/?aid=1988&app_name=tiktok_web&device_platform=web&referer=&root_referer=&user_agent=Mozilla%252F5`, and it returns a response with all desired information! You didn't use a crawl tools like selenium or scrapy and as I known, TikTok don't provide a official API for getting data. What did you do and Why you can get that data, it's seem quite simple ?!!! Waiting for your reply. Thank you so much.

**Comment by davidteather** [comment, marked as answer]
I do use a headless browser (similar to selenium or selenium if you pass use_selenium=True to the constructor) to generate a signature parameter for the URL. This allows the browser to make that request.

You're able to find this in your own browser if you end up going to one of TikTok's sites on your desktop like [#funny](https://www.tiktok.com/tag/funny) and hit f12 on chrome, and go to the network tab in the developer console. From there you can refresh and hit control + f and type in the search box for item_list. This will show you the data TikTok returns from the API.

There's some other strange URL parameters required which the package adds onto and calculates in the case of the _signature parameter.

It does this by launching playwright (similar to selenium) and uses some javascript code to calculate this _signature value which takes into account the browser's user agent, and a few other things.

**Comment by forampatel-xbyte** [comment]
Hi @davidteather 
Is there any way if we can scrape the comment data using the tiktok api?
